doranum/parser.js

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+
+'use strict';
+const Parser = require('../.lib/parser.js');
+
+function parseDOI(doi) {
+  const splited = doi.split('_');
+  return `${splited[0]}.${splited[1]}/${splited[2]}`;
+}
+
+/**
+ * Recognizes the accesses to the platform Doranum
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  let result = {};
+  let path   = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  // let param = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  let match;
+
+  if ((match = /^\/wp-content\/uploads\/[a-z0-9_-]+(10_[0-9]+_([a-z0-9-]+)).zip$/i.exec(path)) !== null) {
+    // /wp-content/uploads/doranum_01_fiche_synthetique_10_13143_mgcn-1863.zip
+    result.rtype    = 'RECORD';
+    result.mime     = 'ZIP';
+    result.doi = parseDOI(match[1]);
```

Comment:
Extracting the things you need directly from the regex should be faster and less prone to side effects

```suggestion
  if ((match = /^\/wp-content\/uploads\/[a-z0-9_-]+10_([0-9]+)_([a-z0-9-]+)\.zip$/i.exec(path)) !== null) {
    // /wp-content/uploads/doranum_01_fiche_synthetique_10_13143_mgcn-1863.zip
    result.rtype = 'RECORD';
    result.mime = 'ZIP';
    result.doi = `10.${match[1]}/${match[2]}`;
```